docs/quick-lint-js.schema.config

```diff
@@ -0,0 +1,65 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+
+    "title": "quick-lint-js configuration",
+    "description": "The quick-lint-js CLI and quick-lint-js editor plugins, can be configured using a quick-lint-js.config file.",
+    "$comment": "Refer https://quick-lint-js.com/config for more details.",
+
+    "type": "object",
+
+    "properties": {
+        "globals": {
+            "description": "Global variables which programs can use. It tells quick-lint-js what global variables to assume exist.",
+            "$comment" : "If the global variables you want are from a popular platform or library, you might want to use 'global-groups' instead.",
+            "type": "object",
+
+            "patternProperties": {
+                "^.*$": {
+                    "anyOf": [
+                        {
+                            "description": "If the value is true, then the variable is defined as if the property’s value was {}. If the value is false, then the variable is not defined, even if a group listed in global-groups would otherwise define the variable.",
+                            "type": "boolean"
+                        },
+                        {
+                            "description": "Specifies whether a global variable can be shadowed or can be assigned to.",
+                            "type": "object",
+
+                            "properties": {
+                                "shadowable": {
+                                    "description": "If true or not present, the variable can redefined in the program’s outer-most scope. If false, the variable cannot be redefined in the program’s outer-most scope.",
```

Comment:
Should we use `"default"` to indicate the default value? https://json-schema.org/draft/2020-12/json-schema-validation.html#name-default